src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -859,10 +860,20 @@ def upload_ssl_cert(resource_group_name, name, certificate_password, certificate
 
     thumb_print = _get_cert(certificate_password, certificate_file)
     cert_name = _generate_cert_name(thumb_print, hosting_environment_profile_param,
-                                    webapp.location, resource_group_name)
+                                    webapp.location, cert_resource_group_name)
     cert = Certificate(password=certificate_password, pfx_blob=cert_contents,
                        location=webapp.location)
-    return client.certificates.create_or_update(resource_group_name, cert_name, cert)
+    return client.certificates.create_or_update(cert_resource_group_name, cert_name, cert)
+
+
+def _get_resource_group_name_by_resource_id(resource_id):
```

Comment:
please replace with `from azure.cli.core.commands.arm import parse_resource_id` and use that common routine